provider/kubernetes/kubernetes.go

```diff
@@ -115,40 +115,36 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 
 	pool.Go(func(stop chan bool) {
 		operation := func() error {
+			stopWatch := make(chan struct{}, 1)
+			defer close(stopWatch)
+			eventsChan, err := k8sClient.WatchAll(p.Namespaces, stopWatch)
+			if err != nil {
+				log.Errorf("Error watching kubernetes events: %v", err)
+				timer := time.NewTimer(1 * time.Second)
```

Comment:
Do you understand why this is required? I mean to populate the error after a second and not directly?

Reply:
@marco-jantke Thank you for review! But after error we exit from function, new iteration will not happen

Reply:
Yes, I agree with that. I was rather asking out of curiosity. That's why I approved right away even though I added the comment.

Reply:
@marco-jantke In fact, this is because, we don't want to send an error if the `pool.Go` was stopped. If we do that, the `backoff.RetryNotify` will continue to retry, and will not stop the `pool.Go`.
So we put a 1s timer that will not trigger in the case if the stoppedChan is populate.